src/components/head-metadata/index.tsx

```diff
@@ -0,0 +1,47 @@
+import HashiHead from '@hashicorp/react-head'
+import { useCurrentProduct } from 'contexts'
+import { HeadMetadataProps } from './types'
+
+/**
+ * Builds up the the necessary meta tags for the site. Rendered in `_app`, where it receives `pageProps.metadata` as props
+ */
+export default function HeadMetadata(props: HeadMetadataProps) {
+  const { name: productName } = useCurrentProduct() ?? {}
+
+  const titleParts = [__config.dev_dot.meta.title]
+  const description = props.description ?? __config.dev_dot.meta.description
+
+  if (productName) {
+    titleParts.unshift(productName)
+  }
+
+  if (props.title) {
+    titleParts.unshift(props.title)
+  }
```

Comment:
I had to look up `unshift` to double-check what it does, could comments be added above each of these with a brief note on the behavior?

```suggestion
  // add `productName` to beginning of `titleParts` array
  if (productName) {
    titleParts.unshift(productName)
  }

  // add `props.title` to beginning of `titleParts` array
  if (props.title) {
    titleParts.unshift(props.title)
  }
```